packages/video_player/ios/Classes/VideoPlayerPlugin.m

```diff
@@ -154,19 +154,27 @@ - (CGAffineTransform)fixTransform:(AVAssetTrack*)videoTrack {
   // At least 2 user videos show a black screen when in portrait mode if we directly use the
   // videoTrack.preferredTransform Setting tx to the height of the video instead of 0, properly
   // displays the video https://github.com/flutter/flutter/issues/17606#issuecomment-413473181
-  if (transform.tx == 0 && transform.ty == 0) {
-    NSInteger rotationDegrees = (NSInteger)round(radiansToDegrees(atan2(transform.b, transform.a)));
-    NSLog(@"TX and TY are 0. Rotation: %ld. Natural width,height: %f, %f", (long)rotationDegrees,
-          videoTrack.naturalSize.width, videoTrack.naturalSize.height);
+  NSInteger rotationDegrees = (NSInteger)round(radiansToDegrees(atan2(transform.b, transform.a)));
+  // This prevents the video to be rendered out of the screen if the metadata contain a rotation but
+  // no translation to compensate the shift induced by the rotation.
+  if (rotationDegrees != 0 && transform.tx == 0 && transform.ty == 0) {
+    NSLog(@"Adding translation to compensate rotation. Rotation = %ld. Natural (width, height) = "
+          @"(%f, %f)",
+          (long)rotationDegrees, videoTrack.naturalSize.width, videoTrack.naturalSize.height);
+    NSLog(@"Uncompensated transform (a, b, c, d, tx, ty) = (%f, %f, %f, %f, %f, %f)", transform.a,
+          transform.b, transform.c, transform.d, transform.tx, transform.ty);
     if (rotationDegrees == 90) {
-      NSLog(@"Setting transform tx");
       transform.tx = videoTrack.naturalSize.height;
       transform.ty = 0;
+    } else if (rotationDegrees == 180) {
```

Comment:
@quentinleguennec Thanks for your PR. I just tested your fix and unfortunately this line is not rendering videos taken at 180°. tx and ty must be 0 for videos in landscape and reverse landscape mode. Furthermore, some videos metadata include incorrect tx, ty values when in portrait mode. In https://github.com/flutter/plugins/pull/1307/files I've added a fix to properly display these videos.

Reply:
It might be because I was using the fixed rotation metadata from this PR:  https://github.com/flutter/plugins/pull/1452
On my side with both those PRs the videos display properly. (even when the video is recorded with a rotation).

Reply:
Have you tried recording your videos with the native Camera app? That's the one I used to record a video in 180° that displays a black screen with tx and ty set to non 0.

Reply:
Good point, I didn't think about checking with the native camera app.
I just did and it is working with the current code when you record a video on your iPhone and play it with the video_player. But it is not working (black screen) if you record the video from the native camera app of an Android an play it with the video_player on an iPhone.

That's because the native iPhone camera app add the translation metadata to the video file (which can be seen with ffprobe). But Android doesn't add it (and VLC doesn't need it either, just iOS). 

With the fix I added it displays the video recorded on Android properly when played on an iPhone. And it also works with a video recorded from an iPhone and played on an iPhone from the video_player.